packages/grafana-ui/src/components/Layout/Grid/Grid.tsx

```diff
@@ -0,0 +1,101 @@
+import { css } from '@emotion/css';
+import React, { useCallback } from 'react';
+
+import { GrafanaTheme2, ThemeSpacingTokens } from '@grafana/data';
+
+import { useStyles2 } from '../../../themes';
+
+export type ItemsAlignment = 'start' | 'end' | 'center' | 'stretch';
+
+export type ContentAlignment = ItemsAlignment | 'space-around' | 'space-between' | 'space-evenly';
+export interface GridProps {
+  children: NonNullable<React.ReactNode>;
+  /** Defines whether the element is a block-lever or a inline-level grid */
+  display?: 'grid' | 'inline-grid';
+  /** Specifies the gutters between columns and between rows. It is overwritten when column or row gap has value */
+  gap?: ThemeSpacingTokens;
+  /** Specifies the gutters between columns */
+  columnGap?: ThemeSpacingTokens;
+  /** Specifies the gutters between rows */
+  rowGap?: ThemeSpacingTokens;
+  /** Defines the columns of the grid */
+  templateColumns?: string;
+  /** Defines the rows of the grid */
+  templateRows?: string;
+  /** Defines the horizontal alignment of the grid items in each cell */
+  justifyItems?: ItemsAlignment;
```

Comment:
Is there a reason we're not using `CSSProperties` interface instead of defining these custom types? 
```suggestion
  justifyItems?: CSSProperties['justifyItems'];
```